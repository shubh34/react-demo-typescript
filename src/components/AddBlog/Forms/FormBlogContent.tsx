import React from 'react';
import { Form, TextArea, TextAreaProps } from 'semantic-ui-react';

interface IFormBlogContent {
	blogContent: string;
	onChange: Function;
	blogContentLabel: string;
}
const FormBlogContent: React.FC<IFormBlogContent> = ({
	blogContent,
	onChange,
	blogContentLabel,
}) => {
	const onBlogChange = (e: React.FormEvent<HTMLTextAreaElement>, data: TextAreaProps) => {
		e.preventDefault();
		onChange('content', data.value);
	};
	return (
		<Form.Field>
			<label>{blogContentLabel}</label>
			<TextArea
				rows={25}
				spellCheck
				name='content'
				placeholder='Enter your content'
				onChange={onBlogChange}
				value={blogContent}
			/>
		</Form.Field>
	);
};

export default FormBlogContent;
