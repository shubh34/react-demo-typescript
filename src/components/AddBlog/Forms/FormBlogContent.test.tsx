import React from 'react';
import { shallow } from 'enzyme';
import FormBlogContent from './FormBlogContent';

describe('FormBlogContent', () => {
	it('renders without crashing', () => {
		const wrapper = shallow(<FormBlogContent blogContent='My Blog content' onChange={jest.fn()} blogContentLabel={'blogContentLabel'} />);
		expect(wrapper).toMatchSnapshot();
	});

	it('should call onChange', () => {
		const onChange = jest.fn();
		const wrapper = shallow(<FormBlogContent blogContent='My Blog Tiltle' onChange={onChange} blogContentLabel={'blogContentLabel'} />);
		wrapper.find('TextArea').simulate('change', { preventDefault: jest.fn() }, { value: 'test' });
		expect(onChange).toBeCalledWith('content', 'test');
	});
});
