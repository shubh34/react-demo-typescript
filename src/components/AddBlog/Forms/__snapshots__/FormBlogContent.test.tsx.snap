// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormBlogContent renders without crashing 1`] = `
<FormField>
  <label>
    blogContentLabel
  </label>
  <TextArea
    as="textarea"
    name="content"
    onChange={[Function]}
    placeholder="Enter your content"
    rows={25}
    spellCheck={true}
    value="My Blog content"
  />
</FormField>
`;
